core/cat/looking_glass/stray_cat.py

```diff
@@ -156,6 +157,40 @@ def send_error(self, error: Union[str, Exception]):
 
         self.__send_ws_json(error_message)
 
+    def recall(
+            self,
+            query: str | List[float],
+            k: int | None = 5,
+            threshold: int | None = None,
+            metadata: dict | None = None,
+            collections: Iterable[MEMORY_COLLECTION] | None = None,
+            override_working_memory: bool = False
+            ):
+
+        if isinstance(query, str):
+            query = self.embedder.embed_query(query)
+
+        if collections:
+            memory_collections = get_args(MEMORY_COLLECTION)
+            collections_not_allowed = [c for c in collections if c not in memory_collections]
+            if any(collections_not_allowed):
+                raise ValueError(
+                    f"Collections not allowed: {', '.join(collections_not_allowed)}. Collections should be any of {', '.join(memory_collections)}")
+        else:
+            collections = self.memory.vectors.collections.keys()
+
+        for collection in collections:
+            vector_memory = getattr(self.memory.vectors, collection)
+            if k:
+                memories = vector_memory.recall_memories_from_embedding(
+                    query, metadata, k, threshold
+                )
+                if override_working_memory:
```

Comment:
overriding is made possible only if the dev defines a `k`. Hence, he or she is not scrolling the whole collection.
Is this ok?